sympy/combinatorics/homomorphisms.py

```diff
@@ -495,33 +497,51 @@ def group_isomorphism(G, H, isomorphism=True):
             raise NotImplementedError("Isomorphism methods are not implemented for infinite groups.")
         _H, h_isomorphism = H._to_perm_group()
 
-    if (g_order != h_order) or (G.is_abelian != H.is_abelian):
-        if not isomorphism:
-            return False
-        return (False, None)
+    if not epimorphism:
+        if (g_order != h_order) or (G.is_abelian != H.is_abelian):
+            if not isomorphism:
+                return False
+            return (False, None)
```

Comment:
If you make your keywords `injective` and `surjective`, than you'll need an `if` statement for each of them and `G.is_abelian != H.is_abelian` could be included as well, though split into parts like this:
```
if injective:
    if (order stuff) or not G.is_abelian and H.is_abelian:
        #false
if surjective:
    if (order stuff) or G.is_abelian and not H.is_abelian:
        #false
```
That's because you can't inject a non-abelian group into an abelian one, or have a surjection from an abelian group to a non-abelian one